### ISSUE TITLE
Calling remove() after a find() query with no match

### ISSUE THREAD

**Issue by xprudhomme** [technosophos/querypath#79]
With previous QueryPath version (2.1.1), I was able to do the following :

```
        $initialXML = XmlUtil::XML_HEADER.'<root><aaa><bbb></bbb><ccc>ddd</ccc></aaa><ddd><eee><bbb/></eee></ddd></root>';

        $qpObject = qp($initialXML)->xpath("//eee");
        $res = $qpObject->remove()->top()->xml();
        var_dump($res);

        $res = $qpObject->xpath("//test")->remove()->top()->xml();
        var_dump($res);
```

Which result was:

string '<?xml version="1.0" encoding="UTF-8"?>&lt;root&gt;&lt;aaa&gt;&lt;bbb/&gt;&lt;ccc&gt;ddd&lt;/ccc&gt;&lt;/aaa&gt;&lt;ddd/&gt;&lt;/root&gt;
' (length=90)

string '<?xml version="1.0" encoding="UTF-8"?>&lt;root&gt;&lt;aaa&gt;&lt;bbb/&gt;&lt;ccc&gt;ddd&lt;/ccc&gt;&lt;/aaa&gt;&lt;ddd/&gt;&lt;/root&gt;
' (length=90)

But now with current version (2.1.2), here is what I got:

string '<?xml version="1.0" encoding="UTF-8"?>&lt;root&gt;&lt;aaa&gt;&lt;bbb/&gt;&lt;ccc&gt;ddd&lt;/ccc&gt;&lt;/aaa&gt;&lt;ddd/&gt;&lt;/root&gt;
' (length=90)

null

Basically, with the new version, if I use the remove method() after a find() query with no match, the result is not the same as before. The only workaround I found is:

```
        $qpObject = qp($initialXML)->xpath("//eee");
        $qpObject->remove();
        $res = $qpObject->top()->xml();
        var_dump($res);

        $qpObject->xpath("//test");
        $qpObject->remove();
        $res = $qpObject->top()->xml();
        var_dump($res);
```

Is this new behavior normal or is it a bug?
